ruler-common/src/main/java/com/spotify/rulercommon/BaseRulerTask.kt

```diff
@@ -0,0 +1,117 @@
+/*
+* Copyright 2023 Spotify AB
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package com.spotify.rulercommon
+
+import com.spotify.ruler.models.AppFile
+import com.spotify.ruler.models.ComponentType
+import com.spotify.rulercommon.apk.ApkParser
+import com.spotify.rulercommon.apk.ApkSanitizer
+import com.spotify.rulercommon.attribution.Attributor
+import com.spotify.rulercommon.dependency.DependencyComponent
+import com.spotify.rulercommon.models.RulerConfig
+import com.spotify.rulercommon.ownership.OwnershipFileParser
+import com.spotify.rulercommon.ownership.OwnershipInfo
+import com.spotify.rulercommon.report.HtmlReporter
+import com.spotify.rulercommon.report.JsonReporter
+import com.spotify.rulercommon.sanitizer.ClassNameSanitizer
+import com.spotify.rulercommon.sanitizer.ResourceNameSanitizer
+import java.io.File
+
+const val FEATURE_NAME = "base"
+
+interface BaseRulerTask {
+
+    fun print(content: String)
+    fun provideMappingFile(): File?
+    fun provideResourceMappingFile(): File?
+    //    fun provideOmitFileBreakdown(): Boolean
```

Comment:
Nit: AFAIK this was moved to Ruler config so we should remove it

Reply:
fixed